### PR TITLE
HPCC-23537 Create debug builds automatically on Docker Hub

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@GordonSmith Please review

File: dockerfiles/action.yml [hunk 1]
```diff
@@ -13,6 +13,9 @@ inputs:
   password:
     description: 'The login password for docker hub'
     required: true
+  buildtype:
+    description: 'The build type - Debug, RelWithDebInfo, or Release'
```
Comment (GordonSmith) [inline dockerfiles/action.yml]: Typo: comma after RelWithDebInfo

File: dockerfiles/buildall.sh [hunk 2]
```diff
@@ -18,6 +18,11 @@ if [[ -n ${INPUT_USERNAME} ]] ; then
   PUSH=1
 fi
 
+if [[ -n ${INPUT_BUILDTYPE} ]] ; then
+  buildtype="--build-arg BUILD_TYPE=$INPUT_BUILDTYPE"
+  suffix=-$INPUT_BUILDTYPE
```
Comment (GordonSmith) [inline dockerfiles/buildall.sh]: The `buildtype` above is not "required", is the intention that blank is "Release"?  If so then the suffix will be different between undefined and Release?

File: dockerfiles/action.yml [hunk 3]
```diff
@@ -13,6 +13,9 @@ inputs:
...
+    required: false
```
Comment (GordonSmith) [inline dockerfiles/action.yml]: Is `required` false correct?

File: dockerfiles/buildall.sh (hunk 2 above)
Comment (richardkchapman) [reply dockerfiles/buildall.sh]: I want no suffix on the default (which is actually RelWithDebInfo), and a suffix only if a build type is explicitly specified.

File: dockerfiles/action.yml (hunk 3 above)
Comment (richardkchapman) [reply dockerfiles/action.yml]: Yes

File: dockerfiles/action.yml (hunk 1 above)
Comment (richardkchapman) [reply dockerfiles/action.yml]: It's an Oxford comma...

File: dockerfiles/platform-build/Dockerfile [hunk 4]
```diff
@@ -17,7 +17,8 @@ WORKDIR /hpcc-dev
 RUN mkdir build
 WORKDIR /hpcc-dev/build
 
-RUN JAVA_HOME=`/usr/libexec/java_home` cmake /hpcc-dev/HPCC-Platform -Wno-dev -DCMAKE_BUILD_TYPE=RelWithDebInfo
+ARG BUILD_TYPE=RelWithDebInfo
```
Comment (richardkchapman) [inline dockerfiles/platform-build/Dockerfile]: This is what sets the default if no build type specified

File: dockerfiles/action.yml (hunk 1 above)
Comment (GordonSmith) [reply dockerfiles/action.yml]: Fair enough

File: dockerfiles/buildall.sh (hunk 2 above)
Comment (GordonSmith) [reply dockerfiles/buildall.sh]: Ok by design,

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Pull request merged**